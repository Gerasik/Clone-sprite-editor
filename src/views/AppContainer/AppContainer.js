export default class AppContainer {
  constructor(className) {
    this.className = className;
    this.block = null;
  }

  render() {
    const container = document.createElement('section');
    container.classList.add(this.className);
    document.body.appendChild(container);
    this.block = container;
  }
}
