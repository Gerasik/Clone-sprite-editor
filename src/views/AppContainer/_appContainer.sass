.container
  width: 100%
  height: 100%
  background-color: #dcdcdc
  // z-index: -1
  display: flex
  align-items: center