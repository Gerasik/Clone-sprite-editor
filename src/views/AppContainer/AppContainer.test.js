import AppContainer from './AppContainer';

const context = {
  className: 'container',
  block: null,
};

describe('new AppContainer', () => {
  const newObj = new AppContainer(context.className, context.parentBlock, context.size);
  it('Should be an instance of Object', () => {
    expect(newObj).toBeInstanceOf(Object);
  });
  it('Should be contain right parameters', () => {
    expect(newObj.className).toBe('container');
    expect(newObj.block).toBeNull();
  });
});

describe('AppContainer.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppContainer.prototype.render).toBeInstanceOf(Function);
  });

  AppContainer.prototype.render.call(context);
  it('Should be render container', () => {
    expect(document.body.innerHTML).toMatchSnapshot();
  });
  it('Block should be not null', () => {
    expect(context.block).not.toBeNull();
  });
});
