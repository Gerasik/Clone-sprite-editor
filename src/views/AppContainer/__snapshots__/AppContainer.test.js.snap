// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer.prototype.render Should be render container 1`] = `"<section class=\\"container\\"></section>"`;
